components/editor/modules/embed/index.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react'
+import { matchBlock } from '../../utils'
+import { findOrCreate } from '../../utils/serialization'
+import MarkdownSerializer from 'slate-mdast-serializer'
+import Loader from '../../../Loader'
+import { colors } from '@project-r/styleguide'
+import { css } from 'glamor'
+import { gql, graphql } from 'react-apollo'
+
+import embedFromUrlPlugin from './embedFromUrlPlugin'
+
+const getEmbed = gql`
+query getEmbed($url: String!) {
+  embed(url: $url) {
+    __typename
+    ... on EmbedInterface {
+      id
+    }
+    ... on Twitter {
+      text
+      userId
+      userName
+      userScreenName
+    }
+    ... on Youtube {
+      userId
+      userName
+      thumbnail
+    }
+    ... on Vimeo {
+      userId
+      userName
+      thumbnail
+    }
+  }
+}
+`
+
+const EmbedLoader =
+  Component =>
+  ({
+    loading,
+    error,
+    embed,
+    data,
+    ...props
+  }) => (
+    <Loader loading={loading} error={error} render={() => {
+      const { node, editor } = props
+      const active = editor.value.blocks.some(
+        block => block.key === node.key
+      )
+      return (
+        <div
+          {...styles.border}
+          data-active={active}
+          contentEditable={false}
+        >
+          <Component
+            embed={data.embed}
```

Comment:
always use the data prop that can also be filled from mdast

Reply:
Can do.